File: ui/src/backend/payloads/admin/loginpayload.ts

export class LoginPayload{
    firstName: string | null;
    lastName: string | null;
    usernameOrEmail: string | null;
    password: string | null;
    token: string | null;
    active: boolean | null = null;
    privileges: Array<string> = new Array<string>();
    
    // Super admin flag
    superAdmin: boolean = false;
    
    // Store info
    posId: number | null;
    posCode: string | null;
    posName: string | null;
    
    // Error handling
    errorCode: string | null;
    errorMessage: string | null;

    constructor(){
        this.firstName = null;
        this.lastName = null;
        this.usernameOrEmail = null;
        this.password = null;
        this.token = null;
        this.superAdmin = false;
        this.posId = null;
        this.posCode = null;
        this.posName = null;
        this.errorCode = null;
        this.errorMessage = null;
    }
}

File: api/src/main/java/com/mystore/manager/api/admin/payload/LoginPayload.java
package com.mystore.manager.api.admin.payload;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotBlank;
import java.util.Set;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
@AllArgsConstructor
@NoArgsConstructor
public class LoginPayload {
    private String firstName;
    private String lastName;

    @NotBlank
    private String usernameOrEmail;

    @NotBlank
    private String password;

    private String token;

    private boolean isActive;
    
    private boolean superAdmin;

    private Set<String> privileges;
    
    // POS info
    private Integer posId;
    private String posCode;
    private String posName;
    
    // Error handling
    private String errorCode;
    private String errorMessage;

    // Epoch-millis timestamp at which the session expires (for frontend session tracking)
    private Long tokenExpiry;

}
